x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java

```diff
@@ -436,6 +442,40 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * For the list of job Ids find all that match existing jobs Ids.
+     * The repsonse is all the job Ids in {@code ids} that match an existing
+     * job Id.
+     * @param ids Job Ids to find 
+     * @param listener The matched Ids listener
+     */
+    public void jobIdMatches(List<String> ids, ActionListener<List<String>> listener) {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        boolQueryBuilder.filter(new TermQueryBuilder(Job.JOB_TYPE.getPreferredName(), Job.ANOMALY_DETECTOR_JOB_TYPE));
+        boolQueryBuilder.filter(new TermsQueryBuilder(Job.ID.getPreferredName(), ids));
+
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder().query(boolQueryBuilder);
+        sourceBuilder.fetchSource(false);
+        sourceBuilder.docValueField(Job.ID.getPreferredName(), DocValueFieldsContext.USE_DEFAULT_FORMAT);
+
+        SearchRequest searchRequest = client.prepareSearch(AnomalyDetectorsIndex.configIndexName())
+                .setIndicesOptions(IndicesOptions.lenientExpandOpen())
+                .setSource(sourceBuilder).request();
```

Comment:
What if `ids.size() > 10`?

Reply:
Good point! I should set `.setSize(ids.size())` 
I'll push that change once the PR build has finished as I'm hoping it will go green